catalogue/engage.py

```diff
@@ -1,162 +1,162 @@
+        files and continue (returns `True`)
+        otherwise, returns `False`
+
+    If the `commit_changes` flag is `True` and the user accepts the offer
+    to commit changes, a new branch is created with the name
+    `"catalogue-%Y%m%d-%H%M%S"` and all tracked files that have been
+    changed & any untracked files will be staged and committed.
+
+    Parameters:
+    ------------
+    repo_path : str
+        path to the code directory
+    catalogue_dir : str
+        path to directory with catalogue output files
+    commit_changes : bool
+        boolean indicating if the user should be prompted to stage and commit changes
+        (optional, default is False)
+
+    Returns:
+    ---------
+    Boolean indicating if git directory is clean
+    """
+
+    try:
+        repo = git.Repo(repo_path, search_parent_directories=True)
+    except InvalidGitRepositoryError:
+        raise InvalidGitRepositoryError("provided code directory is not a valid git repository")
+
+    untracked = prune_files(repo.untracked_files, catalogue_dir)
+
+    if repo.is_dirty() or (len(untracked) != 0):
+        if commit_changes:
+            print("Working directory contains uncommitted changes.")
+            print("Do you want to stage and commit all changes? (y/[n])")
+            user_choice = input().strip().lower()
+            if user_choice == "y" or user_choice == "yes":
+                timestamp = create_timestamp()
+                new_branch = repo.create_head("catalogue-" + timestamp)
+                new_branch.checkout()
+                changed_files = [ item.a_path for item in repo.index.diff(None) ]
+                changed_files.extend(untracked)
+                repo.index.add(changed_files)
+                repo.index.commit("repro-catalogue tool auto commit at " + timestamp)
+                return True
+            elif user_choice == "n" or user_choice == "no" or user_choice == "":
+                return False
+            else:
+                print("Unrecognized response, leaving repository unchaged")
+                return False
+        else:
+            return False
+    else:
+        return True
+
+
+def engage(args):
+    """
+    The `catalogue engage` command.
+
+    The engage command is used prior to running an analysis. The `args` contain
+    paths to `input_data` and `code` (which must be a git repo).
+
+    The engage command:
+        - does a `git_query()` check of the `code` repo
+            (engage does not proceed unless the repo is clean)
+        - gets hashes for the input_data and code (from `construct_dict()`)
+        - saves the hashes to a `.lock` file
+
+    Once engaged (a `.lock` file exists), the command cannot be run again until
+    `disengage` has been run.
+
+    Parameters:
+    ------------
+    args : obj
+        Command line input arguments (argparse.Namespace).
+
+    Returns:
+    ---------
+    None
+    """
+    print(args)

```

Comment:
Why is there a print statement here? I'm guessing it's a left-over from testing?

Reply:
Whoops, I've just fixed this now.